internal/services/nginx/nginx_configuration_resource.go

```diff
@@ -0,0 +1,318 @@
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/validation"
+)
+
+const defaultConfigurationName = "default"
+
+type ConfigFile struct {
+	Content     string `tfschema:"content"`
+	VirtualPath string `tfschema:"virtual_path"`
+}
+
+type ProtectedFile struct {
+	Content     string `tfschema:"content"`
+	VirtualPath string `tfschema:"virtual_path"`
+}
+
+type ConfigurationModel struct {
+	Name              string          `tfschema:"name"` // always default
+	NginxDeploymentId string          `tfschema:"nginx_deployment_id"`
+	ConfigFile        []ConfigFile    `tfschema:"config_file"`
+	ProtectedFile     []ProtectedFile `tfschema:"protected_file"`
+	PackageData       string          `tfschema:"package_data"`
+	RootFile          string          `tfschema:"root_file"`
+}
+
+// ToSDKModel used in both Create and Update
+func (c ConfigurationModel) ToSDKModel() nginxconfiguration.NginxConfiguration {
+	req := nginxconfiguration.NginxConfiguration{
+		Name: pointer.FromString(c.Name),
+		Properties: &nginxconfiguration.NginxConfigurationProperties{
+			RootFile: pointer.FromString(c.RootFile),
+		},
+	}
+
+	var files []nginxconfiguration.NginxConfigurationFile
+	for _, file := range c.ConfigFile {
+		files = append(files, nginxconfiguration.NginxConfigurationFile{
+			Content:     pointer.FromString(file.Content),
+			VirtualPath: pointer.FromString(file.VirtualPath),
+		})
+	}
+	req.Properties.Files = &files
+
+	if len(c.ProtectedFile) > 0 {
+		var protectedFiles []nginxconfiguration.NginxConfigurationFile
+		for _, file := range c.ProtectedFile {
+			protectedFiles = append(protectedFiles, nginxconfiguration.NginxConfigurationFile{
+				Content:     pointer.FromString(file.Content),
+				VirtualPath: pointer.FromString(file.VirtualPath),
+			})
+		}
+		req.Properties.ProtectedFiles = &protectedFiles
+	}
+
+	if c.PackageData != "" {
+		req.Properties.Package = &nginxconfiguration.NginxConfigurationPackage{
+			Data: pointer.FromString(c.PackageData),
+		}
+	}
+
+	return req
+}
+
+type ConfigurationResource struct{}
+
+var _ sdk.Resource = (*ConfigurationResource)(nil)
+
+func (m ConfigurationResource) Arguments() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{
+		"nginx_deployment_id": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: validation.StringIsNotEmpty,
+		},
+
+		"config_file": {
+			Type:     pluginsdk.TypeList,
+			Required: true,
+			Elem: &pluginsdk.Resource{
+				Schema: map[string]*pluginsdk.Schema{
+					"content": {
+						Type:         pluginsdk.TypeString,
+						Required:     true,
+						ValidateFunc: validation.StringIsBase64,
+					},
+
+					"virtual_path": {
+						Type:         pluginsdk.TypeString,
+						Required:     true,
+						ValidateFunc: validation.StringIsNotEmpty,
+					},
+				},
+			},
+		},
+
+		"protected_file": {
```

Comment:
if these files are Protected, presumably they're encrypted? should this be sensitive?

Reply:
sound reasonable. but there is no description or example of this property, so I am not sure if it should be sensitive. I am reaching out to the service team to make sure of it.

Reply:
confirmed as sensitive.